airbyte-db/lib/src/test/java/io/airbyte/db/factory/DataSourceFactoryTest.java

```diff
@@ -0,0 +1,103 @@
+ */
+
+package io.airbyte.db.factory;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.zaxxer.hikari.HikariDataSource;
+import java.util.Map;
+import javax.sql.DataSource;
+import org.junit.jupiter.api.Test;
+import org.postgresql.Driver;
+
+/**
+ * Test suite for the {@link DataSourceFactory} class.
+ */
+public class DataSourceFactoryTest extends AbstractFactoryTest {
+
+  @Test
+  void testCreatingADataSourceWithJdbcUrl() {
+    final String username = container.getUsername();
+    final String password = container.getPassword();
+    final String driverClassName = Driver.class.getName();
+    final String jdbcUrl = container.getJdbcUrl();
+
+    final DataSource dataSource = DataSourceFactory.create(username, password, driverClassName, jdbcUrl);
+    assertNotNull(dataSource);
+    assertEquals(HikariDataSource.class, dataSource.getClass());
+  }
+
+  @Test
+  void testCreatingADataSourceWithJdbcUrlAndConnectionProperties() {
+    final String username = container.getUsername();
+    final String password = container.getPassword();
+    final String driverClassName = Driver.class.getName();
+    final String jdbcUrl = container.getJdbcUrl();
+    final Map<String, String> connectionProperties = Map.of("foo", "bar");
+
+    final DataSource dataSource = DataSourceFactory.create(username, password, driverClassName, jdbcUrl, connectionProperties);
+    assertNotNull(dataSource);
+    assertEquals(HikariDataSource.class, dataSource.getClass());
+  }
+
+  @Test
+  void testCreatingADataSourceWithHostAndPort() {
+    final String username = container.getUsername();
+    final String password = container.getPassword();
+    final String driverClassName = Driver.class.getName();
+    final String host = container.getHost();
+    final Integer port = container.getFirstMappedPort();
+    final String database = container.getDatabaseName();
+
+    final DataSource dataSource = DataSourceFactory.create(username, password, host, port, database, driverClassName);
+    assertNotNull(dataSource);
+    assertEquals(HikariDataSource.class, dataSource.getClass());
+  }
+
+  @Test
+  void testCreatingADataSourceWithHostPortAndConnectionProperties() {
+    final String username = container.getUsername();
+    final String password = container.getPassword();
+    final String driverClassName = Driver.class.getName();
+    final String host = container.getHost();
+    final Integer port = container.getFirstMappedPort();
+    final String database = container.getDatabaseName();
+    final Map<String, String> connectionProperties = Map.of("foo", "bar");
+
+    final DataSource dataSource = DataSourceFactory.create(username, password, host, port, database, driverClassName, connectionProperties);
+    assertNotNull(dataSource);
+    assertEquals(HikariDataSource.class, dataSource.getClass());
```

Comment:
Is it worth also trying to test here that the connectionProperties were set as expected?

Reply:
I will see if I can access those properties to confirm configuration.  Will do this in my other PR that is providing the refactoring.